main/modes/breakout/aabb_utils.h

```diff
@@ -0,0 +1,25 @@
+#ifndef _AABB_UTILS_H_
+#define _AABB_UTILS_H_
+
+#include <stdint.h>
+#include <stdbool.h>
+#include <palette.h>
+
+typedef struct
+{
+    int32_t x;
+    int32_t y;
+} vector_t;
```

Comment:
We do have `vector2d.h` and `geometry.h` but it seems that we all like re-implementing these things anyway.